### PR TITLE
IBC: Centauri migration

### PR DESCRIPTION
**Pull request by blasrodri**
- Point to Strangelove's ibc-go fork, which contains
08-wasm-client and 02-client refactor

- Fix compilation errors brought by the migration

- TODO: ibctypes.MsgTransferResponse does not yet contain
a Sequence attribute

Closes: #XXX

## Author's Checklist

I have...

- [ ] Run and PASSED locally all GAIA integration tests
- [ ] If the change is contentful, I either:
    - [ ] Added a new unit test OR 
    - [ ] Added test cases to existing unit tests
- [ ] OR this change is a trivial rework / code cleanup without any test coverage

If skipped any of the tests above, explain.

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [ ] not applicable
      - [ ] jira ticket `XXX` 
      - [ ] specification (`x/<module>/spec/`) 
      - [ ] README.md 
      - [ ] not documented

### CODE REVIEW

File: x/stakeibc/keeper/msg_server_submit_tx.go [hunk 1]
```diff
@@ -294,7 +297,18 @@ func (k Keeper) SubmitTxs(
...
 	}
 
-	sequence, err := k.ICAControllerKeeper.SendTx(ctx, chanCap, connectionId, portID, packetData, timeoutTimestamp)
+	msg := icacontrollertypes.NewMsgSendTx(owner, connectionId, timeoutTimestamp, packetData)
+
+	msgServer := icacontrollerkeeper.NewMsgServerImpl(&k.ICAControllerKeeper)
+
+	res, err := msgServer.SendTx(ctx, msg)
+
+	if res != nil {
```
Comment (sontrinh16) [inline x/stakeibc/keeper/msg_server_submit_tx.go]: should be err

**Comment by edjroz** [comment]
There seems to be an issue when trying to add gentxs that might be related to `x/genutil`

This was attempted as a local build.
```
strided init apollo --chain-id test
strided keys add ares
strided add-genesis-account $(./strided keys show ares -a) 1000000000stake,1000000000validatortoken
strided gentx ares 1000000000stake --chain-id test

strided collect-gentxs                                                                                                                                       
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x140c166]

goroutine 1 [running]:
github.com/cosmos/cosmos-sdk/x/genutil/types.ValidateAndGetGenTx({0xc0009b4a00, 0x487, 0x488}, 0x2210ae0?, 0x0)
        github.com/cosmos/cosmos-sdk@v0.47.0-rc1/x/genutil/types/genesis_state.go:130 +0x106
github.com/cosmos/cosmos-sdk/x/genutil.CollectTxs({0x448a238, 0xc0002829b0}, 0x1500000?, {0xc000227cb0, 0x6}, {0xc00053ea80, 0x1a}, {{0x1bc8e99, 0xedb803ebb, 0x0}, ...}, ...)
        github.com/cosmos/cosmos-sdk@v0.47.0-rc1/x/genutil/collect.go:114 +0x45e
github.com/cosmos/cosmos-sdk/x/genutil.GenAppStateFromConfig({0x448a238, 0xc0002829b0}, {0x7fb31c51d6a8, 0xc000a0c7c0}, 0xc000c40420, {{0xc000cec9cc, 0x4}, {0xc00053ea80, 0x1a}, {0xc0009f2ae0, ...}, ...}, ...)
        github.com/cosmos/cosmos-sdk@v0.47.0-rc1/x/genutil/collect.go:32 +0x174
github.com/cosmos/cosmos-sdk/x/genutil/client/cli.CollectGenTxsCmd.func1(0xc000b08f00?, {0x58a75c0?, 0x0?, 0x0?})
        github.com/cosmos/cosmos-sdk@v0.47.0-rc1/x/genutil/client/cli/collect.go:53 +0x573
github.com/spf13/cobra.(*Command).execute(0xc000b08f00, {0x58a75c0, 0x0, 0x0})
        github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc000b08900)
        github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.6.1/command.go:968
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        github.com/spf13/cobra@v1.6.1/command.go:961
github.com/cosmos/cosmos-sdk/server/cmd.Execute(0x212c2f0?, {0x0, 0x0}, {0xc000a1a330, 0xd})
        github.com/cosmos/cosmos-sdk@v0.47.0-rc1/server/cmd/execute.go:32 +0x179
```

**Comment by sontrinh16** [comment]
> There seems to be an issue when trying to add gentxs that might be related to `x/genutil`
> 
> This was attempted as a local build.
> 
> ```
> strided init apollo --chain-id test
> strided keys add ares
> strided add-genesis-account $(strided keys show ares -a) 1000000000stake,1000000000validatortoken
> strided gentx ares 1000000000stake --chain-id test
> 
> strided collect-gentxs                                                                                                                                       
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x140c166]
> 
> goroutine 1 [running]:
> github.com/cosmos/cosmos-sdk/x/genutil/types.ValidateAndGetGenTx({0xc0009b4a00, 0x487, 0x488}, 0x2210ae0?, 0x0)
>         github.com/cosmos/cosmos-sdk@v0.47.0-rc1/x/genutil/types/genesis_state.go:130 +0x106
> github.com/cosmos/cosmos-sdk/x/genutil.CollectTxs({0x448a238, 0xc0002829b0}, 0x1500000?, {0xc000227cb0, 0x6}, {0xc00053ea80, 0x1a}, {{0x1bc8e99, 0xedb803ebb, 0x0}, ...}, ...)
>         github.com/cosmos/cosmos-sdk@v0.47.0-rc1/x/genutil/collect.go:114 +0x45e
> github.com/cosmos/cosmos-sdk/x/genutil.GenAppStateFromConfig({0x448a238, 0xc0002829b0}, {0x7fb31c51d6a8, 0xc000a0c7c0}, 0xc000c40420, {{0xc000cec9cc, 0x4}, {0xc00053ea80, 0x1a}, {0xc0009f2ae0, ...}, ...}, ...)
>         github.com/cosmos/cosmos-sdk@v0.47.0-rc1/x/genutil/collect.go:32 +0x174
> github.com/cosmos/cosmos-sdk/x/genutil/client/cli.CollectGenTxsCmd.func1(0xc000b08f00?, {0x58a75c0?, 0x0?, 0x0?})
>         github.com/cosmos/cosmos-sdk@v0.47.0-rc1/x/genutil/client/cli/collect.go:53 +0x573
> github.com/spf13/cobra.(*Command).execute(0xc000b08f00, {0x58a75c0, 0x0, 0x0})
>         github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
> github.com/spf13/cobra.(*Command).ExecuteC(0xc000b08900)
>         github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
> github.com/spf13/cobra.(*Command).Execute(...)
>         github.com/spf13/cobra@v1.6.1/command.go:968
> github.com/spf13/cobra.(*Command).ExecuteContext(...)
>         github.com/spf13/cobra@v1.6.1/command.go:961
> github.com/cosmos/cosmos-sdk/server/cmd.Execute(0x212c2f0?, {0x0, 0x0}, {0xc000a1a330, 0xd})
>         github.com/cosmos/cosmos-sdk@v0.47.0-rc1/server/cmd/execute.go:32 +0x179
> ```
> 
> @sontrinh16

hmmmm im digging into it

**Comment by sontrinh16** [comment]
@edjroz ah fixed, someone initialize genutil app module with empty msg validator

**Comment by edjroz** [comment]
Created a PR #627 with proper static link of the wasmvm and the stride binary, otherwise it's dynamic and when passed on to stage 2 it will fail.

**Comment by asalzmann** [comment]
open

**Pull request closed without merging**